packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -94,13 +168,114 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
   }
 
   @override
-  Future<void> update(Uri golden, Uint8List imageBytes) async {
-    final File goldenFile = _getGoldenFile(golden);
-    await goldenFile.parent.create(recursive: true);
-    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  Uri getTestUri(Uri key, int version) {
+    final String extension = path.extension(key.toString());
```

Comment:
nit: Maybe do the early return before parsing the extension and only stringify key once?
```dart
if (version == null) {
  return key;
}
final String path = key.toString();
final String extension = path.extension(path);
return ...
```